wscript

```diff
@@ -98,6 +98,13 @@ submodules at specific revisions.
             opt.parser.remove_option(k)
             g.add_option(option)
 
+    g.add_option('--apstatedir',
```

Comment:
Is this really needed? Are you going to use it? It seems quite useless to me, especially after you added a custom storage runtime option.

Reply:
I myself, no. This is intended for those wanting to package ArduPilot for their distro (aka to create a .deb, .rpm, flatpak, snap, etc).  This may be different from distro to distro and for all other packages is done on build time time.

The runtime option is intended for users who want to change the default the distro provided. For example to save logs etc on a mountpoint for the sdcard.   The whole issue that motivated this was because of Ubuntu Core not using FHS.

Reply:
> This is intended for those wanting to package ArduPilot for their distro

Who's going to do that except board maintainers? I just think this isn't going to be used by anyone and just brings more complexity to the build system - which is quite messy already.

Reply:
I'm planning to do that. I requested this change because in our distro the default location is read-only.

Reply:
@vilvo I'm confused with that. What you requested in #7983 was a run-time configuration, which is also added by this PR, but is independent from the build time option discussed here.

@lucasdemarchi I think I wasn't clear that I'm not against the AP_STATEDIR macro existing, just adding the option to the build system; as I said, I think it's for a really tiny niche use case that only contributes to increase the build system complexity.

Reply:
It was clear, but I don't agree. They are different level of decisions and follows the same principle of packaging other Linux applications: the option set by devs (the define), the option set by distro packager (the compile-time option) and the option set by the user. The priorities are such that the user has the final say, but we hope everybody will be happy with the default.

I do want ardupilot to be more like other applications on Linux distros, even if for that I have to add a dozen lines in the build system.